test/navigation.spec.js

```diff
@@ -800,6 +800,93 @@ module.exports.describe = function({testRunner, expect, playwright, MAC, WIN, FF
+      ]);
+      expect(messages.join('|')).toBe('route|waitForNavigation|click');
+    });
+    it.fail(FFOX)('clicking anchor should await cross-process navigation', async({page, server}) => {
+      const messages = [];
+      server.setRoute('/empty.html', async (req, res) => {
+        messages.push('route');
+        res.end('done');
+      });
+
+      await page.setContent(`<a href="${server.CROSS_PROCESS_PREFIX + '/empty.html'}">empty.html</a>`);
+
+      await Promise.all([
+        page.click('a').then(() => messages.push('click')),
+        page.waitForNavigation({ waitUntil: [] }).then(() => messages.push('waitForNavigation'))
+      ]);
+      expect(messages.join('|')).toBe('route|waitForNavigation|click');
+    });
+    it.fail(CHROMIUM || FFOX)('should submit form-get that causes navigation', async({page, server}) => {
+      const messages = [];
+      server.setRoute('/empty.html?foo=bar', async (req, res) => {
+        messages.push('route');
+        res.end('done');
+      });
+
+      await page.setContent(`
+        <form action="${server.EMPTY_PAGE}" method="get">
+          <input name="foo" value="bar">
+          <input type="submit" value="Submit">
+        </form>`);
+
+      await Promise.all([
+        page.click('input[type=submit]').then(() => messages.push('click')),
+        page.waitForNavigation({ waitUntil: [] }).then(() => messages.push('waitForNavigation'))
+      ]);
+      expect(messages.join('|')).toBe('route|waitForNavigation|click');
+    });
+    it.fail(CHROMIUM || FFOX)('should submit form-post that causes navigation', async({page, server}) => {
+      const messages = [];
+      server.setRoute('/empty.html', async (req, res) => {
+        messages.push('route');
+        res.end('done');
+      });
+
+      await page.setContent(`
+        <form action="${server.EMPTY_PAGE}" method="post">
+          <input name="foo" value="bar">
+          <input type="submit" value="Submit">
+        </form>`);
+
+      await Promise.all([
+        page.click('input[type=submit]').then(() => messages.push('click')),
+        page.waitForNavigation({ waitUntil: [] }).then(() => messages.push('waitForNavigation'))
+      ]);
+      expect(messages.join('|')).toBe('route|waitForNavigation|click');
+    });
+    it('should not throw when clicking on links which do not commit navigation', async({page, server, httpsServer}) => {
+      await page.goto(server.EMPTY_PAGE);
+      await page.setContent(`<a href='${httpsServer.EMPTY_PAGE}'>foobar</a>`);
+      await page.click('a');
+    });
+    it('should not throw when clicking on download link', async({page, server, httpsServer}) => {
+      await page.setContent(`<a href="${server.PREFIX}/wasm/table2.wasm" download=true>table2.wasm</a>`);
+      await page.click('a');
+    });
+    it.fail(CHROMIUM || FFOX)('should not hang on window.stop', async({page, server, httpsServer}) => {
```

Comment:
I don't feel comfortable landing this patch until this test passes.

Reply:
That's too bad, because it does not pass without my patch either.

Reply:
Figured it out!

Reply:
Interesting! Something stalled the evaluation?